src/buyflow/Buyflow.tsx

```diff
@@ -0,0 +1,42 @@
+import React, { useState } from 'react'
+import AgeStep from './AgeStep'
+import EmailStep from './EmailStep'
+import SummaryStep from './SummaryStep'
+
+interface BuyflowProps {
+  productId: ProductIds
+}
+
+export enum ProductIds {
+  devIns = 'dev_ins',
+}
+
+const PRODUCT_IDS_TO_NAMES = {
+  [ProductIds.devIns]: 'Developer Insurance',
+}
+
+const Buyflow: React.FC<BuyflowProps> = (props) => {
+  const [currentStep, setStep] = useState('email')
+  const [collectedData, updateData] = useState({
+    email: '',
+    age: 0,
+  })
+  const getStepCallback = (nextStep: string) => (field: string, value: any) => {
+    updateData({ ...collectedData, [field]: value })
+    setStep(nextStep)
+  }
```

Comment:
we would still use a callback for `...Step` components to use, but since next step can be calculated automatically we won't need `nextStep` argument:
```suggestion
  const stepCallback = useCallback((field: string, value: any) => {
    updateData(prevData => { ...prevData, [field]: value })
    goToNextStep()
  }, [goToNextStep])
```